### ISSUE TITLE
Patroller Thresholds

### ISSUE THREAD

**Issue by ToMadoRe** [strands-project/autonomous_patrolling#102]
Anyone has an explanation why I discovered our robot standing on the docking station this morning with a battery level of 50% even though we set the threshold to
`gen.add("low_battery", int_t, 0, "low battery treshold", 65, 55, 95)` in /autonomous_patrolling/waypoint_patroller/cfg/PatrollerTresholds.cfg?

Also, is it normal that the charging takes about 5 hours for battery_level to get from 50% to ~95%?

**Comment by bfalacerda** [comment]
Did you edit it before the run?

**Comment by ToMadoRe** [comment]
I did. But wasn't there during the docking phase as it was during night.

**Comment by bfalacerda** [comment]
then it might to due with those values being overwritten somewhere, I'm not sure. Run rqt_Reconfigure and check what are the values for the thresholds for the patroller. If you have 65 there, then getting to 50 is probably some other  problem

**Comment by ghost** [comment]
It might be the same problem as we have, where the robot is on the charging
station while not charging. Then it simply goes down and never undocks. But
we also found that rqt reconfigure is pretty much the only way to set the
values nicely unless you rebuild everything after setting the values in the
config file.

**Comment by bfalacerda** [comment]
you're having docking failures @AlexanderHermans ? It-s been pretty reliable here and since there's no recovery for that now, maybe you should try to recalibrate, as that is a big point of failure I think

**Comment by ghost** [comment]
No no, it's not that the docking fails, that is actually only happing
rarely. But the charging unit in the robot has a problem. Metralabs has
been informed and told us they would send us a new one to swap it. Not sure
what is going on there, but sometimes it just doesn't charge, even though
the charging light is on. An indication for this is also if the current is
positive (1~3A) while on the charging station.

**Comment by bfalacerda** [comment]
oh yeah, forgot about that problem

**Comment by ToMadoRe** [comment]
Well, can't test the parameters(using rqt_Reconfigure) again before the marathon. So, I hope everything goes fine when resetting it to the original thresholds. 
i am just wondering why the parameters shouldn't have been updated when I had changed them before recompiling the catkin_ws. Anyway...
@AlexanderHermans: What is your problem with the docking? We also had that issue that our robot was not charging on the docking station last week. For some reason, the contacts of the robot were too high and didn't touch the metal plates on the docking station any more. Therefore we had to put a cardboard below the docking station and now everything is fine so far.
